lib/core/io.js

```diff
@@ -225,8 +224,8 @@ class FetchReader extends Reader {
 	async init() {
 		super.init();
 		if (isHttpFamily(this.url) && !this.preventHeadRequest) {
-			const response = await sendFetchRequest(HTTP_METHOD_HEAD, this.url, this.options);
-			this.size = Number(response.headers.get(HTTP_HEADER_CONTENT_LENGTH));
+			const response = await sendFetchRequest(HTTP_METHOD_GET, this.url, this.options, { range: "bytes=0-0"});
+			this.size = Number(response.headers.get(HTTP_HEADER_CONTENT_RANGE).split('/')[1]);
```

Comment:
`response.headers.get(HTTP_HEADER_CONTENT_RANGE)` can return `undefined` and cause this code to throw an error.

Reply:
Also, the HTTP server might not support the header "Content-Range" or nor return the size of the content, see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Range. These cases should be taken into account.

Reply:
`{ range: "bytes=0-0"}` must be replaced with `[[ "range", "bytes=0-0"]]`

Reply:
Done.